dotnet/src/webdriver/PinnedScript.cs

```diff
@@ -17,83 +17,70 @@
 // under the License.
 // </copyright>
 
-using System;
 using System.Globalization;
 
+#nullable enable
+
 namespace OpenQA.Selenium
 {
     /// <summary>
     /// A class representing a pinned JavaScript function that can be repeatedly called
     /// without sending the entire script across the wire for every execution.
     /// </summary>
-    public class PinnedScript
+    public sealed class PinnedScript
     {
-        private string scriptSource;
-        private string scriptHandle;
-        private string scriptId;
-
         /// <summary>
         /// Initializes a new instance of the <see cref="PinnedScript"/> class.
         /// </summary>
         /// <param name="script">The body of the JavaScript function to pin.</param>
+        /// <param name="stringHandle">The unique handle for this pinned script.</param>
+        /// <param name="scriptId">The internal ID of this script.</param>
         /// <remarks>
         /// This constructor is explicitly internal. Creation of pinned script objects
         /// is strictly the perview of Selenium, and should not be required by external
         /// libraries.
         /// </remarks>
-        internal PinnedScript(string script)
+        internal PinnedScript(string script, string stringHandle, string scriptId)
         {
-            this.scriptSource = script;
-            this.scriptHandle = Guid.NewGuid().ToString("N");
+            this.Source = script;
+            this.Handle = stringHandle;
+            this.ScriptId = scriptId;
         }
 
         /// <summary>
         /// Gets the unique handle for this pinned script.
         /// </summary>
-        public string Handle
-        {
-            get { return this.scriptHandle; }
-        }
+        public string Handle { get; }
```

Comment:
Can it be field?

Reply:
No, it is property:
![image](https://github.com/user-attachments/assets/19fac2c0-be3c-466c-a844-41b0a7e545ae)